packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart

```diff
@@ -279,6 +280,9 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface<
       );
     }
 
+    // Dependencies for AnalyticsMetadataType
+    _stateMachine.addInstance(EndpointInfoStoreManager());
```

Comment:
```suggestion
    _stateMachine.addBuilder<EndpointInfoStoreManager>(EndpointInfoStoreManager.new);
```

Reply:
Updated the dart and flutter level to use this.  

```
stateMachine.addBuilder<EndpointInfoStoreManager>(
      FlutterEndpointInfoStoreManager.new,
    );

_stateMachine
        .addBuilder<EndpointInfoStoreManager>(EndpointInfoStoreManager.new);
```

Reply:
If you update both, then we arrive at the same issue. What gets created in the code below?

```dart
stateMachine.addBuilder<EndpointInfoStoreManager>(FlutterEndpointInfoStoreManager.new);
stateMachine.addBuilder<EndpointInfoStoreManager>(EndpointInfoStoreManager.new);
final endpointManager = stateMachine.getOrCreate<EndpointInfoStoreManager>();
```

Reply:
Of course.. thanks for pointing that out. 

I understand `getOrCreate` better now.